### PR TITLE
Add Rollback option to resource group level resources

### PR DESCRIPTION
**Pull request by chidmdxx**
## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/ResourceManager/Resources/Commands.Resources/ChangeLog.md [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 ## Version 6.4.1
 * Fixed issue with default resource groups not being set.
+* Update New-AzureRmResourceGroupDeployment with new parameter RollbackAction
```
Comment (maddieclayton) [inline src/ResourceManager/Resources/Commands.Resources/ChangeLog.md]: Please move this to the "Current Release" header to ensure this will be released.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
 {
+    using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
```
Comment (maddieclayton) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs]: Please move this with the rest of the "using" statements

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs [hunk 3]
```diff
@@ -45,6 +47,9 @@ public class NewAzureResourceGroupDeploymentCmdlet : ResourceWithParameterCmdlet
         [ValidateSet("RequestContent", "ResponseContent", "All", "None", IgnoreCase = true)]
         public string DeploymentDebugLogLevel { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of a deployment to roll back to on error, or use as a flag to roll back to the last successful deployment.")]
```
Comment (maddieclayton) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs]: Please regenerate all the help files to reflect this change: https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md

File: src/ResourceManager/Resources/Commands.Resources/ChangeLog.md [hunk 4]
```diff
@@ -14,7 +14,7 @@
         - Additional information about change #4
 
     ## YYYY.MM.DD - Version X.Y.Z (Previous Release)
-    * Overview of change #1
+    * Overview of change 1#
```
Comment (maddieclayton) [inline src/ResourceManager/Resources/Commands.Resources/ChangeLog.md]: Revert this change

**Comment by maddieclayton** [comment]
Please either link your review or fill out a new design review from here: https://github.com/Azure/azure-powershell-cmdlet-review-pr

**Comment by chidmdxx** [comment]
@maddieclayton filled the following design review https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/146

**Comment by cormacpayne** [comment]
@chidmdxx Hey Rachid, for some reason, GitHub isn't letting me pull in the latest from the `preview` branch into your branch, so would you mind doing so to resolve the build error that you're currently seeing?

**Comment by chidmdxx** [comment]
@cormacpayne rebased the branch with the latest from upstream

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs [hunk 5]
```diff
@@ -45,6 +45,9 @@ public class NewAzureResourceGroupDeploymentCmdlet : ResourceWithParameterCmdlet
...
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs]: This needs to have an argument completer that allows the user to choose 'LastSuccessfulDeployment'  - A user selecting this is the way you should choose to send 'LastSuccessful' in the code below

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/TestAzureResourceGroupDeploymentCmdlet.cs [hunk 6]
```diff
@@ -35,6 +35,9 @@ public class TestAzureResourceGroupDeploymentCmdlet : ResourceWithParameterCmdle
         [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment mode.")]
         public DeploymentMode Mode { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The name of a deployment to roll back to on error, or use as a flag to roll back to the last successful deployment.")]
+        public string RollbackAction { get; set; }
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/TestAzureResourceGroupDeploymentCmdlet.cs]: Why would this go under test deployment?  How does this impact a deployment test?

Comment (chidmdxx) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/TestAzureResourceGroupDeploymentCmdlet.cs]: deployment validation will check for the validity of the specified on error deployment

File: src/ResourceManager/Resources/Commands.Resources/help/Test-AzureRmResourceGroupDeployment.md [hunk 7]
```diff
@@ -153,6 +161,36 @@ Accept pipeline input: True (ByPropertyName)
...
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
 ### -TemplateFile
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.Resources/help/Test-AzureRmResourceGroupDeployment.md]: remove any change to AzureRM.psm1 below

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs [hunk 8]
```diff
@@ -45,6 +46,12 @@ public class NewAzureResourceGroupDeploymentCmdlet : ResourceWithParameterCmdlet
         [ValidateSet("RequestContent", "ResponseContent", "All", "None", IgnoreCase = true)]
         public string DeploymentDebugLogLevel { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Rollback to the last successful deployment in the resource group, should not be present if -RollBackDeploymentName is used.")]
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs]: these need to be in separate parameter sets - any parameter that is not in any parameter set is in all parameter sets.  I can put a PR into your branch to show how this is done.

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs [hunk 9]
```diff
@@ -66,6 +73,11 @@ public override void ExecuteCmdlet()
                 ResourceGroupName,
                 () =>
                 {
+                    if (RollbackToLastDeployment && !string.IsNullOrEmpty(RollBackDeploymentName))
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/NewAzureResourceGroupDeploymentCmdlet.cs]: Instead, we shoudl do this with parameter sets,

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/TestAzureResourceGroupDeploymentCmdlet.cs [hunk 10]
```diff
@@ -35,19 +37,37 @@ public class TestAzureResourceGroupDeploymentCmdlet : ResourceWithParameterCmdle
         [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment mode.")]
         public DeploymentMode Mode { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "Rollback to the last successful deployment in the resource group, should not be present if -RollBackDeploymentName is used.")]
```
Comment (markcowl) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/ResourceGroupDeployments/TestAzureResourceGroupDeploymentCmdlet.cs]: same comment
